reddit_decider/__init__.py

```diff
@@ -379,40 +379,42 @@ def expose(
     def get_variant_for_identifier(
         self,
         experiment_name: str,
-        identifier: str,
+        user_id: str = "",
```

Comment:
I think I'd instead leave identifier as-is, and add an identifier_type, which you can restrict to user_id, device_id, canonical_url.   Doing this would let you not have to worry about what to do with the user_id field if it isn't supplied, and it will mean that if the call is, eg., setting device_id, only experiments that care about device_id will care about the different setup.

Reply:
good call